### PR TITLE
Make calculateFrictionFactor evalution  + critical bugfix in MSW friction model

### PR DESCRIPTION
**Pull request by totto82**
This PR depends on OPM/opm-material#337

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this opm-material=337 please

File: opm/simulators/wells/MSWellHelpers.hpp [hunk 1]
```diff
@@ -133,31 +133,31 @@ namespace mswellhelpers
...
-        const double re = std::abs(diameter * w / (area * mu));
+        const ValueType re = Opm::abs( diameter * w / (area * mu));
 
         if ( re == 0.0 ) {
             // make sure it is because the mass rate is zero
             assert(w == 0.);
             return 0.0;
         }
 
-        const double re_value1 = 200.;
```
Comment (GitPaean) [inline opm/simulators/wells/MSWellHelpers.hpp]: Thanks for catching this.

**Comment by alfbr** [comment]
This may impact performance significantly. Will do performance testing and report back.

**Comment by alfbr** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test | Configuration | Relative |
| ---- | ------------- | -------- |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=400

**Comment by atgeirr** [comment]
> This may impact performance significantly. Will do performance testing and report back.

I cannot see the benchmark results, but maybe we should split the bugfix into a separate PR?

**Comment by totto82** [comment]
I think this should be merged as is. It clearly fixes two bugs. And spend the time finding the reminding bugs in the code that causes the convergence issues these and other test results indicates.

**Comment by alfbr** [comment]
> I cannot see the benchmark results

Still some teething problems I am afraid, expect a permanent fix soon :-) Testing proceeding independently on a wider selection of models.

**Comment by alfbr** [comment]
> I think this should be merged as is.

We have no problem with that as long as the effort will be followed through to completion.

**Comment by GitPaean** [comment]
It happened before.  Due to other existing bugs, some positive correction might impact the result in a negative way.  

I am okay with either we merge this PR now or wait for more bug fixing to make sure the PR has a positive impact.

**Comment by atgeirr** [comment]
Ok, it should be merged as is then. We must update the reference data first, that is expected and only applies to the multisegment well tests.

**Comment by atgeirr** [comment]
jenkins build this opm-material=337 update_data please

**Comment by atgeirr** [comment]
jenkins build this opm-material=337 update_data please

**Comment by atgeirr** [comment]
jenkins build this update_data please

**Comment by GitPaean** [comment]
There are two parts in this PR, one is to make the `calculateFrictionFactor` carries the correct derivatives, the other one is change a bug in the formulation `200` to `2000`. 

Further testing shows mostly likely it is the first part that impacts the performance.

**Comment by atgeirr** [comment]
Looks like tons of tests got updated in the opm-tests PR, that is surprising! Are these accumulated changes over several PRs that were too small to trigger test failures? I think that might be it, as the test failures here remain the three expected ones. Any comments to that @totto82 before I merge?

**Comment by totto82** [comment]
Only "the three" expected fails locally. Could it be an issue with the update? Run it again?

**Comment by atgeirr** [comment]
Rerunning changed nothing. I think I'll trust in the system and merge.

**Comment by atgeirr** [comment]
jenkins build this opm-tests=123 please

**Comment by GitPaean** [comment]
Closing OPM/opm-tests#123 for one hour or so to see whether it makes any difference.  Jenkins have been having problem in updating existing PRs.

**Comment by GitPaean** [comment]
jenkins build this update_data please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=123 please

**Comment by atgeirr** [comment]
Merged as discussed yesterday.

**Pull request merged**